libfreerdp/common/settings.c

```diff
@@ -2048,7 +2048,11 @@ BOOL freerdp_settings_set_string_from_utf16N(rdpSettings* settings, size_t id, c
 
 	char* str = ConvertWCharNToUtf8Alloc(param, length, &len);
 	if (!str && (length != 0))
-		return FALSE;
+	{
+		const size_t wlen = _wcsnlen(param, length);
```

Comment:
Don't quite get this additional check ... is it for checking empty strings?
Shouldn't this be correctly handled by `ConvertWCharNToUtf8Alloc`?

Reply:
this case handles the return value for the following problem:
if `length > 0` but the string contained in `param` is empty then `ConvertWCharNToUtf8Alloc` returns `NULL` with `len=0`

`ConvertWCharNToUtf8Alloc` does correctly convert the string, but we do not want to fail here as the conversion was successful and just clear the (maybe) existing value from settings.

Reply:
That's what I mean - if we pass an empty string to `ConvertWCharNToUtf8Alloc` shouldn't it also return an empty (non-null) string? IIRC there are some cases with string settings that differentiate between `NULL` and `""` - like you said in this case we would clear the value (which should be done in case `NULL` is passed) and not set it to an empty string.

Reply:
ok, removed for the time being, the change might result in another pr.